.travis.yml

```diff
@@ -7,54 +7,76 @@ cache:
     - secure: "Hv6Sj29d9gI63bORCGHVeX5KqEryAAXf9INQL1ABs33a46wVzGCHM/9H/mE3FItHN7RZaeOEeUoDSePOt/8NBtZSAZ1PvbqhMgztvSWtweS9ezzClUHxN3BRZPSyNmZdrK0S9fo+NXgcgqcmOM9gQzrAduHIkWOOwdSGWgT6zNlAZovddwJ8RUtKavZ7U4szJyMtzlQ8S2K8PdTX9glH1U+8ZLHr54uDrhl3WqsCoBEV0GV59eOIaIseDorCEscVj1EDYCPdkYdU6/AWiuUlqXeOvC67CLcu/5W8m05YWXrEhKG+17jEWRPM+G+Kiyj6HccvjTTaXqTsx9ZLuWmSt86mgo15DmJlVtzpvkptgHhheDyp5MkifValXN3gvWzk3VqjHIUh81dP2o8iAWrGfJcYg+8hdv/obfNGvfcSKwgdefIp4B6f/VRvIySvHUqKavj2fEAGDMYHqahHyKGF/VegHuQbCT+F+mI+GdAwun7XzuDVAlftnbrl2kPq7Kjs+ahU4ZIU4pk6eY15rh/WPlR2v0M7O4nfv1IoyHh9F/G6foqwqDoT+RBF1VMr9WDEezz5B35si6ILI/+SG/7BnbHkFRGorOw86ZqMoIs7dkdpz8uHiz+ba83COuASbrHUMfTG3HQ65XY7puqccSPBQzg7yoVmqPUJJNkaroVMWmU="
 
-# Some of the instructions to install and run minikube are from https://blog.travis-ci.com/2017-10-26-running-kubernetes-on-travis-ci-with-minikube
 before_install:
+  # download things concurrently
+  - |
+    $(
+      curl -Lo dep "https://github.com/golang/dep/releases/download/${DEP_VERSION}/dep-linux-amd64" &
+      curl -Lo install_bazel.sh "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh" &
+      curl -Lo helm.tar.gz "https://storage.googleapis.com/kubernetes-helm/helm-${HELM_VERSION}-linux-amd64.tar.gz" &
+      curl -Lo kubectl "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl" &
+      wait
+    )
   - |
-    curl -Lo dep "https://github.com/golang/dep/releases/download/${DEP_VERSION}/dep-linux-amd64"
     chmod +x dep
     sudo mv dep /usr/local/bin/
   - |
-    curl -Lo install.sh "https://github.com/bazelbuild/bazel/releases/download/${BAZEL_VERSION}/bazel-${BAZEL_VERSION}-installer-linux-x86_64.sh"
-    chmod +x install.sh
-    ./install.sh --user
-    rm -f install.sh
+    chmod +x install_bazel.sh
+    ./install_bazel.sh --user
+    rm -f install_bazel.sh
     mv .bazelrc.travis .bazelrc
   - |
-    curl -Lo kubectl "https://storage.googleapis.com/kubernetes-release/release/${KUBECTL_VERSION}/bin/linux/amd64/kubectl"
+    tar -zxvf helm.tar.gz
+    sudo mv linux-amd64/helm /usr/local/bin/
+    rm -rf linux-amd64
+  - |
     chmod +x kubectl
     sudo mv kubectl /usr/local/bin/
-  - |
-    go get sigs.k8s.io/kind
-    docker pull kindest/node:${KUBE_VERSION}
-    kind create cluster --image="kindest/node:${KUBE_VERSION}"
 install:
   - make setup-ci
 
 script:
   - make update-bazel
   - make verify
   - make test-ci
+  - go get sigs.k8s.io/kind
```

Comment:
This now runs after unit tests and verification - they are most common source of failures and also run reasonably fast.